### ISSUE TITLE
[flake8] Errors show up in ALEInfo but not in quickfix or in gutter

### ISSUE THREAD

**Issue by MarcelRobitaille** [dense-analysis/ale#3401]
## Information

**VIM version**

```
NVIM v0.5.0-752-g759a05407
Build type: RelWithDebInfo
```

Operating System: Arch Linux

## What went wrong

I opened a python file with badly formatted code. `:ALEInfo` shows errors are output from flake8,
but nothing is showing up in quickfix or in the gutter.

Something went wrong in specifically this place, and I also searched through both open and closed issues for the same problem before reporting a bug here.

## Reproducing the bug

1. Configure Ale to use flake8 for python
1. Open python buffer with badly formatted code
1. Observe that no errors are shown in the gutter

### :ALEInfo

```

 Current Filetype: python
Available Linters: ['bandit', 'flake8', 'mypy', 'prospector', 'pycodestyle', 'pydocstyle', 'pyflakes', 'pylama', 'pylint', 'pyls', 'pyre', 'pyright', 'vulture']
  Enabled Linters: ['flake8']
  Ignored Linters: []
 Suggested Fixers: 
  'add_blank_lines_for_python_control_statements' - Add blank lines before control statements.
  'autopep8' - Fix PEP8 issues with autopep8.
  'black' - Fix PEP8 issues with black.
  'isort' - Sort Python imports with isort.
  'remove_trailing_lines' - Remove all blank lines at the end of a file.
  'reorder-python-imports' - Sort Python imports with reorder-python-imports.
  'trim_whitespace' - Remove all trailing whitespace characters at the end of every line.
  'yapf' - Fix Python files with yapf.
 Linter Variables:

let g:ale_python_auto_pipenv = 0
let g:ale_python_flake8_auto_pipenv = 0
let g:ale_python_flake8_change_directory = 'project'
let g:ale_python_flake8_executable = 'flake8'
let g:ale_python_flake8_options = ''
let g:ale_python_flake8_use_global = 0
 Global Variables:

let g:ale_cache_executable_check_failures = v:null
let g:ale_change_sign_column_color = 0
let g:ale_command_wrapper = ''
let g:ale_completion_delay = v:null
let g:ale_completion_enabled = 0
let g:ale_completion_max_suggestions = v:null
let g:ale_disable_lsp = 0
let g:ale_echo_cursor = 1
let g:ale_echo_msg_error_str = 'Error'
let g:ale_echo_msg_format = '%code: %%s'
let g:ale_echo_msg_info_str = 'Info'
let g:ale_echo_msg_warning_str = 'Warning'
let g:ale_enabled = 1
let g:ale_fix_on_save = 0
let g:ale_fixers = {'*': ['remove_trailing_lines', 'trim_whitespace'], 'javascript': ['eslint'], 'python': ['autopep8']}
let g:ale_history_enabled = 1
let g:ale_history_log_output = 1
let g:ale_keep_list_window_open = 0
let g:ale_lint_delay = 200
let g:ale_lint_on_enter = 1
let g:ale_lint_on_filetype_changed = 1
let g:ale_lint_on_insert_leave = 1
let g:ale_lint_on_save = 1
let g:ale_lint_on_text_changed = 'normal'
let g:ale_linter_aliases = {}
let g:ale_linters = {'jsx': ['eslint'], 'c': ['gcc'], 'tex': ['chktex'], 'cpp': ['g++'], 'python': ['flake8']}
let g:ale_linters_explicit = 0
let g:ale_linters_ignore = {}
let g:ale_list_vertical = 0
let g:ale_list_window_size = 10
let g:ale_loclist_msg_format = '%code: %%s'
let g:ale_lsp_root = {}
let g:ale_max_buffer_history_size = 20
let g:ale_max_signs = -1
let g:ale_maximum_file_size = v:null
let g:ale_open_list = 0
let g:ale_pattern_options = v:null
let g:ale_pattern_options_enabled = v:null
let g:ale_set_balloons = 0
let g:ale_set_highlights = 1
let g:ale_set_loclist = 1
let g:ale_set_quickfix = 0
let g:ale_set_signs = 1
let g:ale_sign_column_always = 0
let g:ale_sign_error = ''
let g:ale_sign_info = ''
let g:ale_sign_offset = 1000000
let g:ale_sign_style_error = ''
let g:ale_sign_style_warning = ''
let g:ale_sign_warning = ''
let g:ale_sign_highlight_linenrs = 0
let g:ale_statusline_format = v:null
let g:ale_type_map = {}
let g:ale_use_global_executables = v:null
let g:ale_virtualtext_cursor = 0
let g:ale_warn_about_trailing_blank_lines = 1
let g:ale_warn_about_trailing_whitespace = 1
  Command History:

(executable check - success) flake8
(finished - exit code 0) ['/bin/zsh', '-c', '''flake8'' --version']

<<<OUTPUT STARTS>>>
3.8.4 (mccabe: 0.6.1, pycodestyle: 2.6.0, pyflakes: 2.2.0) CPython 3.8.6 on
Linux
<<<OUTPUT ENDS>>>

(finished - exit code 1) ['/bin/zsh', '-c', 'cd ''/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions'' && ''flake8'' --format=default --stdin-display-name ''/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py'' - < ''/tmp/nvimhWiFHw/6/a5p5.py''']

<<<OUTPUT STARTS>>>
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:14:1: E302 expected 2 blank lines, found 1
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:18:5: F841 local variable 'initial_total_value' is assigned to but never used
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:23:27: F821 undefined name 'initial'
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:30:41: F821 undefined name 'initial'
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:35:1: E302 expected 2 blank lines, found 1
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:55:1: E305 expected 2 blank lines after class or function definition, found 1
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:56:12: E231 missing whitespace after ','
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:56:14: E231 missing whitespace after ','
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:56:17: E231 missing whitespace after ','
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:56:20: E231 missing whitespace after ','
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:56:22: E231 missing whitespace after ','
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:56:25: E231 missing whitespace after ','
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:56:28: E231 missing whitespace after ','
/home/marcel/Nextcloud/Waterloo/Courses/ECE606: Algorithm Design and Analysis/Assignments/Assignment 5/Solutions/a5p5.py:56:30: E231 missing whitespace after ','
<<<OUTPUT ENDS>>>
```

**Comment by w0rp** [comment]
The regular expression for parsing the flake8 errors will need to be updated. It's probably tripping up on the `ECE606: ` in the filename, and needs to be a bit less greedy.

**Comment by hsanson** [comment]
Removing the ":" from the folder name would allow the handler to parse the errors.  I think is difficult to create a regexp that would work with the current folder names since the ":" are used to mark the different parts of the error message.
